trin-core/src/utils/provider.rs

```diff
@@ -112,31 +121,39 @@ fn build_infura_ws_url_from_env() -> String {
     format!("{INFURA_BASE_WS_URL}{infura_project_id}")
 }
 
-fn get_geth_client_id_from_env() -> String {
-    match env::var("GETH_CLIENT_ID") {
+fn get_pandaops_client_id_from_env() -> String {
+    match env::var("PANDAOPS_CLIENT_ID") {
         Ok(val) => val,
         Err(_) => panic!(
-            "Must supply Geth client id as environment variable, like:\n\
-            GETH_CLIENT_ID=\"your-key-here\" trin"
+            "Must supply pandaops client id as environment variable, like:\n\
+            PANDAOPS_CLIENT_ID=\"your-key-here\" trin"
         ),
     }
 }
 
-fn get_geth_client_secret_from_env() -> String {
-    match env::var("GETH_CLIENT_SECRET") {
+fn get_pandaops_client_secret_from_env() -> String {
+    match env::var("PANDAOPS_CLIENT_SECRET") {
         Ok(val) => val,
         Err(_) => panic!(
-            "Must supply Geth client secret as environment variable, like:\n\
-            GETH_CLIENT_SECRET=\"your-key-here\" trin"
+            "Must supply pandaops client secret as environment variable, like:\n\
+            PANDAOPS_CLIENT_SECRET=\"your-key-here\" trin"
         ),
     }
 }
 
-fn build_geth_http_client_from_env(geth_url: String) -> ureq::Request {
-    let client_id = get_geth_client_id_from_env();
-    let client_secret = get_geth_client_secret_from_env();
-    ureq::post(&geth_url)
+fn build_pandaops_http_client_from_env(pandaops_url: String) -> ureq::Request {
+    let client_id = get_pandaops_client_id_from_env();
+    let client_secret = get_pandaops_client_secret_from_env();
+    ureq::post(&pandaops_url)
         .set("Content-Type", "application/json")
         .set("CF-Access-Client-Id", client_id.as_str())
         .set("CF-Access-Client-Secret", client_secret.as_str())
 }
+
+fn build_local_node_http_client(node_url: String) -> ureq::Request {
+    let url = match node_url.starts_with("127.0.0.1") {
```

Comment:
I'd also use the [url](https://docs.rs/url/2.3.1/url/index.html) library here

Reply:
done :)